engine/api/src/main/java/io/deephaven/engine/table/Table.java

```diff
@@ -427,126 +427,6 @@ public interface Table extends
-                case GREATER_THAN_EQUAL:
-                    return Table.AsOfMatchRule.GREATER_THAN_EQUAL;
-                case GREATER_THAN:
-                    return Table.AsOfMatchRule.GREATER_THAN;
-            }
-            throw new IllegalStateException("Unexpected rule " + rule);
-        }
-    }
-
-    /**
-     * Looks up the columns in the rightTable that meet the match conditions in the columnsToMatch list. Matching is
-     * done exactly for the first n-1 columns and via a binary search for the last match pair. The columns of the
-     * original table are returned intact, together with the columns from rightTable defined in a comma separated list
-     * "columnsToAdd"
-     *
-     * @param rightTable The right side table on the join.
-     * @param columnsToMatch A comma separated list of match conditions ("leftColumn=rightColumn" or
-     *        "columnFoundInBoth")
-     * @param columnsToAdd A comma separated list with the columns from the left side that need to be added to the right
-     *        side as a result of the match.
-     * @return a new table joined according to the specification in columnsToMatch and columnsToAdd
-     */
-    Table aj(Table rightTable, MatchPair[] columnsToMatch, MatchPair[] columnsToAdd, AsOfMatchRule asOfMatchRule);
-
-    /**
-     * Looks up the columns in the rightTable that meet the match conditions in the columnsToMatch list. Matching is
-     * done exactly for the first n-1 columns and via a binary search for the last match pair. The columns of the
-     * original table are returned intact, together with the columns from rightTable defined in a comma separated list
-     * "columnsToAdd"
-     *
-     * @param rightTable The right side table on the join.
-     * @param columnsToMatch A comma separated list of match conditions ("leftColumn=rightColumn" or
-     *        "columnFoundInBoth")
-     * @param columnsToAdd A comma separated list with the columns from the left side that need to be added to the right
-     *        side as a result of the match.
-     * @return a new table joined according to the specification in columnsToMatch and columnsToAdd
-     */
-    Table aj(Table rightTable, MatchPair[] columnsToMatch, MatchPair[] columnsToAdd);
-
-    /**
-     * Looks up the columns in the rightTable that meet the match conditions in the columnsToMatch list. Matching is
-     * done exactly for the first n-1 columns and via a binary search for the last match pair. The columns of the
-     * original table are returned intact, together with all the columns from rightTable.
-     *
-     * @param rightTable The right side table on the join.
-     * @param columnsToMatch A comma separated list of match conditions ("leftColumn=rightColumn" or
-     *        "columnFoundInBoth")
-     * @return a new table joined according to the specification in columnsToMatch and columnsToAdd
-     */
-    Table aj(Table rightTable, Collection<String> columnsToMatch);
```

Comment:
The unfortunate part about these joins is that the stringy versions are all single strings;

```java
L.naturalJoin(R, "A,B", "C,D")
// instead of
// L.naturalJoin(R, ["A", "B"], ["C", "D"])
```

I don't know if there's a good way to improve this situation without more drastic changes.

The good news is that the python api can make a much prettier wrapper in these regards.